src/main/java/com/ebay/myriad/scheduler/MyriadDriver.java

```diff
@@ -56,8 +62,29 @@ public MyriadDriver(final MyriadScheduler scheduler,
             throw new RuntimeException(e);
         }
 
-        this.driver = new MesosSchedulerDriver(scheduler,
-                frameworkInfoBuilder.build(), cfg.getMesosMaster());
+        String mesosAuthenticationPrincipal = cfg.getMesosAuthenticationPrincipal();
+        String mesosAuthenticationSecretFilename = cfg.getMesosAuthenticationSecretFilename();
+        if (StringUtils.isNotEmpty(mesosAuthenticationPrincipal)) {
+            frameworkInfoBuilder.setPrincipal(mesosAuthenticationPrincipal);
+
+            Credential.Builder credBuilder = Credential.newBuilder();
```

Comment:
Would rather see this called "credentialBuilder" instead of "credBuilder", but I wouldn't hold back the merge for it.